experimental/ir/fdp.go

```diff
@@ -410,34 +400,30 @@ type syntheticNames map[intern.ID]struct{}
 
 func (sn *syntheticNames) generate(candidate string, message Type) string {
```

Comment:
This seems like a good candidate for a simple unit test.

Reply:
Can't test this directly yet because constructing a Type manually is not something that's really supported. I added a test for the string manipulation part though.